<a href="https://colab.research.google.com/github/RachmadAprisandhy/PembelajaranMesin/blob/main/JS02/JS02-Assignmet.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
!pip install -q opencv-python matplotlib

persiapan

In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.preprocessing import LabelEncoder, StandardScaler

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


memuat data

In [6]:
path = '/content/drive/MyDrive/Pembelajaran mesin/wbc.csv'
df = pd.read_csv(path)

inspeksi data

In [7]:
df.shape

(569, 33)

tugas
Pada tugas pratikum ini Anda akan menggunakan data "Wisconsin Breast Cancer". Data tersebut terdiri dari 569 data yang digunakan untuk mendiagnonis jenis kanker Malignant (M) dan Benign (B). Tugas Anda adalah,

1. Pisahkan antara variabel yang dapat digunakan dan variabel yang tidak dapat digunakan.

2. Lakukan proses encoding pada kolom "diagnosis".

3. Lakukan proses standardisasi pada semua kolom yang memiliki nilai numerik.


1. Pisahkan antara variabel yang dapat digunakan dan variabel yang tidak dapat digunakan.

info data

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 33 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   id                       569 non-null    int64  
 1   diagnosis                569 non-null    object 
 2   radius_mean              569 non-null    float64
 3   texture_mean             569 non-null    float64
 4   perimeter_mean           569 non-null    float64
 5   area_mean                569 non-null    float64
 6   smoothness_mean          569 non-null    float64
 7   compactness_mean         569 non-null    float64
 8   concavity_mean           569 non-null    float64
 9   concave points_mean      569 non-null    float64
 10  symmetry_mean            569 non-null    float64
 11  fractal_dimension_mean   569 non-null    float64
 12  radius_se                569 non-null    float64
 13  texture_se               569 non-null    float64
 14  perimeter_se             5

pemisahan data

In [9]:
df_clean = df.drop(columns=["id", "Unnamed: 32"])

pisahkan data diagnosis kanker

In [19]:
y = df_clean["diagnosis"]
X = df_clean.drop(columns=["diagnosis"])

print("Ukuran X (Fitur):", X.shape)
print("Ukuran y (Target):", y.shape)

jumlah_kolom_X = X.shape[1]
print("Jumlah kolom X:", jumlah_kolom_X)

jumlah_kolom_y = df_clean[["diagnosis"]].shape[1]
print("Jumlah kolom y:", jumlah_kolom_y)

Ukuran X (Fitur): (569, 30)
Ukuran y (Target): (569,)
Jumlah kolom X: 30
Jumlah kolom y: 1


Lakukan proses encoding pada kolom "diagnosis".

In [22]:
le = LabelEncoder()
df['diagnosis'] = le.fit_transform(df['diagnosis'])


cek hasil encoding

In [23]:
print(y.head())

0    M
1    M
2    M
3    M
4    M
Name: diagnosis, dtype: object


In [24]:
df.head()

,id,diagnosis,radius_mean,texture_mean,perimeter_mean,area_mean,smoothness_mean,compactness_mean,concavity_mean,concave points_mean,...,texture_worst,perimeter_worst,area_worst,smoothness_worst,compactness_worst,concavity_worst,concave points_worst,symmetry_worst,fractal_dimension_worst,Unnamed: 32
0,842302,1,17.99,10.38,122.80,1001.0,0.11840,0.27760,0.3001,0.14710,...,17.33,184.60,2019.0,0.1622,0.6656,0.7119,0.2654,0.4601,0.11890,NaN
1,842517,1,20.57,17.77,132.90,1326.0,0.08474,0.07864,0.0869,0.07017,...,23.41,158.80,1956.0,0.1238,0.1866,0.2416,0.1860,0.2750,0.08902,NaN
2,84300903,1,19.69,21.25,130.00,1203.0,0.10960,0.15990,0.1974,0.12790,...,25.53,152.50,1709.0,0.1444,0.4245,0.4504,0.2430,0.3613,0.08758,NaN
3,84348301,1,11.42,20.38,77.58,386.1,0.14250,0.28390,0.2414,0.10520,...,26.50,98.87,567.7,0.2098,0.8663,0.6869,0.2575,0.6638,0.17300,NaN
4,84358402,1,20.29,14.34,135.10,1297.0,0.10030,0.13280,0.1980,0.10430,...,16.67,152.20,1575.0,0.1374,0.2050,0.4000,0.1625,0.2364,0.07678,NaN


Lakukan proses standardisasi pada semua kolom yang memiliki nilai numerik.

In [28]:
from sklearn.preprocessing import StandardScaler


# X berisi 30 kolom numerik murni
X = df.drop(columns=["id", "diagnosis", "Unnamed: 32"], errors="ignore")

# 2. Ambil daftar kolom numerik dari X saja
kolom_numerik = X.select_dtypes(include=["int64", "float64"]).columns

# 3. Lakukan standardisasi pada X
scaler = StandardScaler()
X_scaled = X.copy()
X_scaled[kolom_numerik] = scaler.fit_transform(X[kolom_numerik])

# Tampilkan hasil
print("Kolom yang distandardisasi (Total 30):")
print(kolom_numerik.tolist())

print("\nHasil Standardisasi (X_scaled):")
print(X_scaled.head())

Kolom yang distandardisasi (Total 30):
['radius_mean', 'texture_mean', 'perimeter_mean', 'area_mean', 'smoothness_mean', 'compactness_mean', 'concavity_mean', 'concave points_mean', 'symmetry_mean', 'fractal_dimension_mean', 'radius_se', 'texture_se', 'perimeter_se', 'area_se', 'smoothness_se', 'compactness_se', 'concavity_se', 'concave points_se', 'symmetry_se', 'fractal_dimension_se', 'radius_worst', 'texture_worst', 'perimeter_worst', 'area_worst', 'smoothness_worst', 'compactness_worst', 'concavity_worst', 'concave points_worst', 'symmetry_worst', 'fractal_dimension_worst']

Hasil Standardisasi (X_scaled):
   radius_mean  texture_mean  perimeter_mean  area_mean  smoothness_mean  \
0     1.097064     -2.073335        1.269934   0.984375         1.568466   
1     1.829821     -0.353632        1.685955   1.908708        -0.826962   
2     1.579888      0.456187        1.566503   1.558884         0.942210   
3    -0.768909      0.253732       -0.592687  -0.764464         3.283553   
4 